### ISSUE TITLE
Problem running Auto-WEKA: NULL

### ISSUE THREAD

**Issue by areeberg** [larskotthoff/autoweka#31]
Hi, 
I´m trying to run autoWEKA however I always get the same error. The output log only contain: 12:58:47: Started Auto-WEKA for myproject
12:59:08: null
I´m using a macos El capitan, with java 8 update 111. I tried with attribute as class as well.

**Comment by larskotthoff** [comment]
Thanks for the report, this is an issue that has been reported before and that I've been unable to pin down. Could you try with a different JVM please and see if you get the same result?

**Comment by areeberg** [comment]
Hi Lars, 
I have too much thins that depend on this java version, so I didnt change the JVM. However i did some tests, and got an interesting result. 
I tried to uso auto weka with Weka 3.8 and 3.7, however still got the same error. 
But when calling by terminal, it worked. I used the command from your tutorial (java -cp autoweka.jar weka.classifiers.meta.AutoWEKAClassifier -t Tinput2.arff -timeLimit 15 -no-cv).
This way, probably the error is when dealing with exceptions with Weka interaction. 
Another doubt that I have is how to get the cost and gamma parameters from SVM using the AutoWEKAClassifier (or how to call the SMO classifier). The answer that I got only shows the arguments for a random forest algorithm. 
Thank you very much.

**Comment by larskotthoff** [comment]
Interesting, thanks for testing.

Auto-WEKA doesn't show parameters for all learners, only for the one it identified as the best.

**Comment by areeberg** [comment]
Hum.. but is there any way to force it use a specific classifier?

**Comment by larskotthoff** [comment]
No. We are planning to add this in one of the future releases, but at the moment you can't do this.

**Comment by areeberg** [comment]
Oh, ok. Thank you for your time!

**Comment by larskotthoff** [comment]
Oh and if you don't mind doing some more testing, could you check whether this happens when using the Auto-WEKA panel and/or using Auto-WEKA classifier through the classifiers panel?

**Comment by areeberg** [comment]
I got the same errors with both panels. 
![autowekapanel](https://cloud.githubusercontent.com/assets/7761867/20174478/1b4a27f0-a725-11e6-9cf2-fbd62c45ba41.png)
![classifypanel](https://cloud.githubusercontent.com/assets/7761867/20174479/1b4aec26-a725-11e6-864a-8b06c1711af6.png)

**Comment by larskotthoff** [comment]
Thanks a lot, I'll investigate.

**Comment by namankumar** [comment]
Hello, wondering if there is an update on this? I'm experiencing the same issue.

**Comment by larskotthoff** [comment]
I'm currently working to get GUI access to a Mac where I can test this. I haven't forgotten about it, it's just taking longer than expected :)

**Comment by namankumar** [comment]
Great! Do let me know if I can be of help, I'm eager to try out AutoWeka!

**Comment by namankumar** [comment]
An update on this - the issue is caused by AutoWEKAClassifier looking java in the Weka.app but not finding it. See error log below. I don't have java installed except what ships with Weka. I'm running macOS Sierra and I get the same error whether through the GUI or through the CLI.

`2016-12-10 16:37:29.989 [Thread-13] ERROR AutoWEKAClassifier - Cannot run program "/Users/naman/automl/weka/weka-3-8-0/weka-3-8-0-oracle-jvm.app/Contents/PlugIns/jdk1.8.0_66.jdk/Contents/Home/jre/bin/java": error=2, No such file or directory
`

**Comment by larskotthoff** [comment]
Thanks for the follow-up, will investigate.

**Comment by hek26** [comment]
I am running into a similar issue. However, I noticed that if I run Auto-Weka on the example data sets on the Auto-Weka site - it actually runs just fine. I am running it on OS X El Capitan and Weka 3.8.0. It only gives me this error on my own data sets.

**Comment by larskotthoff** [comment]
Could you post one of those datasets here please? Do other classifiers in WEKA run ok with it?

**Comment by hek26** [comment]
I've tried the following from this page: http://www.cs.ubc.ca/labs/beta/Projects/autoweka/datasets/

dexter, car, german credit

- they all seem to work. 

I have also attached a truncated version of one of my own data sets. 
[Example.txt](https://github.com/automl/autoweka/files/669434/Example.txt)

**Comment by larskotthoff** [comment]
Your dataset works fine for me.

**Comment by hek26** [comment]
Interesting...Please do let us know if you come up with an update. Thanks!

**Comment by larskotthoff** [comment]
The issue and a workaround should be documented in #34 now. Could you please try the current master and see if you get a sensible error message?

**Comment by hek26** [comment]
Thanks for getting back to us on this issue. I used the package manager to uninstall and re-install AutoWeka and then subsequently tested it again, it does not seem to work. It returns the same error 'null.'

**Comment by larskotthoff** [comment]
Ah, sorry, I haven't made a new release yet. I've put the new package at http://www.cs.ubc.ca/~larsko/autoweka.zip -- you can install it from there directly with the package manager or download first.

**Comment by hek26** [comment]
No worries. I have installed it and now I get the following: 

20:57:01: Java executable could not be found. Please refer to "Known Issues" in the Auto-WEKA manual.

Is this a Java version issue?

**Comment by larskotthoff** [comment]
I've attached the new manual -- have a look at section 4 for a description of the issue.

There's nothing we can do about that in Auto-WEKA itself, but please let me know if the workaround in the manual helps at all.

[manual.pdf](https://github.com/automl/autoweka/files/720842/manual.pdf)

**Comment by namankumar** [comment]
New file along with the instructions in the new manual are working great. Thanks Lars!

**Comment by larskotthoff** [comment]
This should work fine now, so closing here. @hek26 please reopen if the problem persists for you.

**Comment by ecamy** [comment]
I also encounter the problem like hek26,  when I run auto-weka ,show the error like this:
Java executable could not be found
I use weka3-8-1, and java 1.7, would you help me?

**Comment by larskotthoff** [comment]
What version of Auto-WEKA are you running?

**Comment by ecamy** [comment]
Hi, Dear Lars,
I tried auto-weka V2.3 and V2.2 in Win8 OS, they have the same error. For finding the reason, I do the following:
 
1.I also run weka from command line
2.change  dataset into standard weka
3.chagne  classifier, such as J48 and Part
 
Still all the same. I copied the window and the log for you as the two attachments, please check. Thanks a lot!
 
Amy Wang
At 2017-03-06 18:03:57, "Lars Kotthoff" <notifications@github.com> wrote:


What version of Auto-WEKA are you running?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.
18:26:33: Weka Explorer
18:26:33: (c) 1999-2016 The University of Waikato, Hamilton, New Zealand
18:26:33: web: http://www.cs.waikato.ac.nz/~ml/weka/
18:26:33: Started on Monday, 6 March 2017
18:30:02: Base relation is now Apache-weka.filters.unsupervised.attribute.Discretize-B5-M-1.0-R10-weka.filters.unsupervised.attribute.NumericToNominal-R1-weka.filters.unsupervised.attribute.NumericToNominal-R2-weka.filters.unsupervised.attribute.NumericToNominal-R3-weka.filters.unsupervised.attribute.NumericToNominal-R4-weka.filters.unsupervised.attribute.NumericToNominal-R5-weka.filters.unsupervised.attribute.NumericToNominal-R6-weka.filters.unsupervised.attribute.NumericToNominal-R7-weka.filters.unsupervised.attribute.NumericToNominal-R8-weka.filters.unsupervised.attribute.NumericToNominal-R9 (192 instances)
18:30:02: java.lang.NullPointerException
18:30:11: Started Auto-WEKA for Apache-weka.filters.unsupervised.attribute.Discretize-B5-M-1.0-R10-weka.filters.unsupervised.attribute.NumericToNominal-R1-weka.filters.unsupervised.attribute.NumericToNominal-R2-weka.filters.unsupervised.attribute.NumericToNominal-R3-weka.filters.unsupervised.attribute.NumericToNominal-R4-weka.filters.unsupervised.attribute.NumericToNominal-R5-weka.filters.unsupervised.attribute.NumericToNominal-R6-weka.filters.unsupervised.attribute.NumericToNominal-R7-weka.filters.unsupervised.attribute.NumericToNominal-R8-weka.filters.unsupervised.attribute.NumericToNominal-R9
18:30:11: Java executable could not be found. Please refer to "Known Issues" in the Auto-WEKA manual.
18:30:30: Started Auto-WEKA for Apache-weka.filters.unsupervised.attribute.Discretize-B5-M-1.0-R10-weka.filters.unsupervised.attribute.NumericToNominal-R1-weka.filters.unsupervised.attribute.NumericToNominal-R2-weka.filters.unsupervised.attribute.NumericToNominal-R3-weka.filters.unsupervised.attribute.NumericToNominal-R4-weka.filters.unsupervised.attribute.NumericToNominal-R5-weka.filters.unsupervised.attribute.NumericToNominal-R6-weka.filters.unsupervised.attribute.NumericToNominal-R7-weka.filters.unsupervised.attribute.NumericToNominal-R8-weka.filters.unsupervised.attribute.NumericToNominal-R9
18:30:30: Java executable could not be found. Please refer to "Known Issues" in the Auto-WEKA manual.

**Comment by larskotthoff** [comment]
It looks like you don't have a JDK installed. Please install that.

**Comment by ecamy** [comment]
My jdk installed: 1.7.0_25, and these classifiers work well using any data set except auto-weka .



Today I will load it into eclipse to try it again. Thanks a lot.  


At 2017-03-06 20:21:09, "Lars Kotthoff" <notifications@github.com> wrote:

It looks like you don't have a JDK installed. Please install that.

**Comment by larskotthoff** [comment]
Hmm, it sounds like it should work then :)

Can you give me some more information on your setup please? What operating system are you running, and how exactly are you running WEKA and Auto-WEKA?

**Comment by ecamy** [comment]
Hi,Lars, Unfortunately, my auto-weka still can't work till now, my steps are the follows:


1.  download auto-weka2.3 from http://www.cs.ubc.ca/labs/beta/Projects/autoweka/autoweka-2.3.zip
2.  run weka3.8.1 GUI(in win8 OS), Tools -> PakageManager ->  click File/URL -> select autoweka-2.3/zip -> install, then restart. 
3.  Explorer -> Preprocess -> open file -> select dataset ->  classify -> use training set -> Auto-weka -> start


then display " Java executable could not be found. Please refer to "known Issues" in the Auto-WEKA manual".


Then I imported the project into eclipse,   still the same. I haven't found the error String in the project. 


In fact, I have programmed on weka 3-8-1 recently.  Someday I saw your paper about auto-weka, I want to try the strong function, and see whether to use it or not in the future.


Now I am afraid I must lay it aside for a while, because anther thing not finished. 


Thanks a lot.


Amy Wang


At 2017-03-07 18:31:28, "Lars Kotthoff" <notifications@github.com> wrote:


Hmm, it sounds like it should work then :)

Can you give me some more information on your setup please? What operating system are you running, and how exactly are you running WEKA and Auto-WEKA?

**Comment by larskotthoff** [comment]
Sorry, I'll have a look as soon as I get a chance.

**Comment by leobavila** [comment]
Hi, larskotthoff . Did you find any hint about this error in evaluating the classifier auto-weka? I have the same problem here: Problem running Auto-Weka - NULL.
I'm working on Win10 and I have downloaded the github package and installed through the Package Manager.
Thanks in advance.

**Comment by larskotthoff** [comment]
Sorry no, still can't reproduce this. Have you tried the released version instead of the github version?

**Comment by leobavila** [comment]
Yes, I've tried to run the zip file "Auto-WEKA 2.6 (July 2017) WEKA package" version using the Package Manager. Would you recommend any special care with settings?

- Windows 10
- Weka Version: weka-3-8-3jre-x64
- Java Version: java version "1.8.0_181" Java(TM) SE Runtime Environment (build 1.8.0_181-b13) Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

link: https://www.cs.ubc.ca/labs/beta/Projects/autoweka/#software

**Comment by mik3hall** [comment]
If the issue actually is that the java command executable is missing I think this can possibly be worked around a couple ways. There was some discussion on the Weka mailing list. I can only find a cached version of the archives for that discussion now. 

https://webcache.googleusercontent.com/search?q=cache:OnObG0s2VewJ:https://list.waikato.ac.nz/pipermail/wekalist/2017-June/070112.html+&cd=8&hl=en&ct=clnk&gl=us&client=safari

My solution for OS X was to manually include the java command executable in the Weka application bundle's embedded java jdk. I believe the above link also mentions possible properties or environment variables that can be used to point to the executable. I'm not sure those solutions worked with the application or command line though. 

For what it's worth. I had brought this up on one of the jdk mailing lists when they were considering a jep to improve the javapackager command for making desktop applications. They had agreed, I think, to include an option to include all the bin executable commands, or none. They did not go forward on that JEP.

Currently I believe support for javapackager is being dropped from the javafx group. Eventually to be picked up  by the jdk core-dev group with another jep, or it might be the same jep updated. 

http://openjdk.java.net/jeps/343

It sounds like javafx will stop supporting javapackager at java 10, so for java 11 there will be no jdk provided desktop application tool  to begin with.

Anyhow, this is my understanding from still sort of lurking on some of the jdk lists. I did add core-dev just to see if any discussion comes up there.

**Comment by leobavila** [comment]
I was able to run it in a Oracle Virtual Box using Ubuntu, and I couldn't fix it for Windows.

Further, I have tried to test the whole set of datasets used in the paper (Auto-WEKA: Combined Selection and Hyperparameter Optimization of Classification Algorithms) but I had the Auto-Weka: NULL problem for some of them (cifar-10, cifar-10 small, dorothea, and kdd cup), even in the virtual box using Ubuntu!

Do I have to perform any pre-processing to the data before using the tool? It's very strange I can use the tool for some datasets and I can't for others.

I have loaded each dataset into Weka and then I used the Auto-Weka tab to evaluate the models, setting memory and time limits. That is it, right?
